pkg/proxy/proxy.go

```diff
@@ -30,17 +30,21 @@ var (
 	ErrFileToStringEmpty error = errors.New("file is empty")
 )
 
+type loadTestSpecCreator func(*http.Request, *zap.Logger) (apisLoadTestV1.LoadTestSpec, error)
+
 // Proxy handler
 type Proxy struct {
-	maxLoadTestsRun int
-	kubeClient      *kube.Client
+	maxLoadTestsRun     int
+	kubeClient          *kube.Client
+	httpToSpecConverter loadTestSpecCreator
```

Comment:
No need to inject this internal function here

Reply:
I find the current implementation more transparent. It allows me to mock this dependency in tests in a straightforward way.